File: Difficulty: Easy/Reverse a linked list/reverse-a-linked-list.java
/*
class Node {
    int data;
    Node next;

    Node(int x) {
        data = x;
        next = null;
    }
}
*/

class Solution {
    Node reverseList(Node head) {
        Node curr = head;
        Node next = null;
        Node prev = null;
        
        while (curr != null) {
            next = curr.next;
            curr.next = prev;
            
            prev = curr;
            curr = next;
        }
        
        return prev;
    }
}

File: Difficulty: Easy/Reverse a linked list/reverse-a-linked-list.py
"""
class Node:
    def __init__(self, val):
        self.data = val
        self.next = None
"""

class Solution:
    def reverseList(self, head):
        stack = []
        
        curr = head
        while curr.next:
            stack.append(curr)
            curr = curr.next
        
        head = curr
        
        while stack:
            curr.next = stack.pop()
            curr = curr.next
        curr.next = None

        return head
        